src/backend/users.js

```diff
@@ -0,0 +1,24 @@
+const router = require('express').Router();
+
+router.get('/users', (req, res) => {
+    req.app.get('UsersService').callbackMeTheListing((result, err) => {
+        req.app.get('ErrorHandler').returnResponse(result, err, res);
+    });
+});
+
+router.get('/users/:userId', (req, res) => {
+    const userId = parseInt(req.params.userId);
+    req.app.get('UsersService').callbackMeTheOne(userId, (result, err) => {
+        req.app.get('ErrorHandler').returnResponse(result, err, res);
```

Comment:
Pomysł na umieszczenie error handlera na końcu każdego wywołania bardzo dobry. Powinno się tak robić.

Wykonanie trochę gorsze, bo odciąłeś się od promisów, które pozwalają to zrobić o wiele sprawniej. 

```js
router.get('/users/:userId', (req, res) => {
  const userId = parseInt(req.params.userId);
  doStuffBasedOnArgumentsAndReturnPromise(userId)
    .then((result) => { res.send("success and stuff") })
    .catch(printErrorsTo(res))
})

printErrorsTo(res) zwraca funkcję przyjmującą jeden argument - error. Każdy error jaki poleci gdziekolwiek w łańcuchu promisów (wliczjąc w to thena tuż ponad) trafi do tej funkcji i będzie mógł być wyświetlony